packages/ui/src/components/va-data-table/VaDataTable.vue

```diff
@@ -41,104 +39,118 @@
-                  <span v-if="$props.grid" class="va-data-table__grid-column-header">{{ columnsComputed[cellIndex].label }}</span>
-                  {{ cell.value }}
-                </slot>
-              </td>
-            </tr>
-          </transition-group>
-
-          <slot name="bodyAppend" />
+              <template
+                v-for="(row, index) in renderBuffer"
+                :key="`table-row_${uniqueKey(row, index)}`"
+              >
+                <tr
+                  class="va-data-table__table-tr"
+                  :class="[{ selected: isRowSelected(row), 'va-data-table__table-tr--expanded': row.isExpandableRowVisible }]"
+                  v-bind="getRowBind(row)"
+                  @click="onRowClickHandler('row:click', $event, row)"
+                  @dblclick="onRowClickHandler('row:dblclick', $event, row)"
+                  @contextmenu="onRowClickHandler('row:contextmenu', $event, row)"
+                >
+                  <td
+                    v-if="selectable && !$props.grid"
+                    class="va-data-table__table-td va-data-table__table-cell-select"
+                    :key="`selectable_${uniqueKey(row, index)}`"
+                    @selectstart.prevent
+                  >
+                    <va-checkbox
+                      class="va-data-table__table-cell-checkbox"
+                      :model-value="isRowSelected(row)"
+                      :color="selectedColor"
+                      :aria-label="t(`selectRowByIndex`, { index: row.initialIndex })"
+                      @click.shift.exact.stop="shiftSelectRows(row)"
+                      @click.ctrl.exact.stop="ctrlSelectRow(row)"
+                      @click.exact.stop="ctrlSelectRow(row)"
+                    />
+                  </td>
+
+                  <td
+                    v-for="(cell, cellIndex) in row.cells"
+                    :key="`table-cell_${cell.column.name + cell.rowIndex}`"
+                    class="va-data-table__table-td"
+                    :class="getClass(cell.column.tdClass)"
+                    :style="[getCellCSSVariables(cell), getStyle(cell.column.tdStyle)]"
+                    v-bind="getCellBind(cell, row)"
+                  >
+                    <slot
+                      v-if="`cell(${cell.column.name})` in $slots"
+                      :name="`cell(${cell.column.name})`"
+                      v-bind="{ ...cell, row, isExpanded: row.isExpandableRowVisible }"
+                    />
+
+                    <slot v-else name="cell" v-bind="{ cell, row }">
+                      <span v-if="$props.grid" class="va-data-table__grid-column-header">{{ columnsComputed[cellIndex].label }}</span>
+                      {{ cell.value }}
+                    </slot>
+                  </td>
+                </tr>
+                <tr
+                  v-show="row.isExpandableRowVisible"
+                  class="va-data-table__table-tr--expanded va-data-table__table-expanded-content"
```

Comment:
Also, classes here are important to have, because likely user want table to be customizable.